### PR TITLE
refactor(ts): only generate tsconfig.json if missing

### LINKED ISSUE #5338: Using @nuxt/typescript overrides any existing tsconfig.json file
### Version

[v2.5.1](https://github.com/nuxt.js/releases/tag/v2.5.1)

### Reproduction link

[https://codesandbox.io/s/0qmykr7wq0](https://codesandbox.io/s/0qmykr7wq0)

### Steps to reproduce

Using the official Nuxt with TypeScript example on CodeSandbox:
- Adjust the `tsconfig.json` in any way
- Restart the sandbox

TS support will be detected again and any modifications to `tsconfig.json` will be overridden.

### What is expected ?

The existing `tsconfig.json` is unmodified.

### What is actually happening?

The `tsconfig.json` content is overridden.

### Additional comments?

I've also noticed that the generation of the `tsconfig.json` doesn't seem to respect the `srcDir` setting in the Nuxt config.

For some background: we have a setup where we use TS for our server-side code and our Nuxt code in the same repo. At present, we have a `tsconfig.json` in the root for our server code and then a separate `tsconfig.json` for our Nuxt setup in a subdirectory (which is ignored by the first config file).

Since 2.5.0 our root `tsconfig.json` is being overridden by the automatically generated one. I don't think this would cause any issues if it respected the `srcDir` config option (i.e. generated it in the subdirectory) although it seems feasible to me that people may want to modify their config and overriding it every time may not be the best option.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8900">#c8900</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by kevinmarrec**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Fixes #5338 
Closes #5356 

Makes Nuxt not overriding `tsconfig.json` anymore, and only generates it with defaults if its missing.

Furthermore, TypeScript support is now only initialized if `@nuxt/typescript` is installed, so no more issues for projects which have a `tsconfig.json` in their root folder but doesn't want Nuxt to enable its TypeScript support.

The initializing state of TypeScript Support (includes `tsconfig.json` autogeneration step) is also more clear now :

Screenshot from 2019-03-26 10-42-18 

## Migration Guide

If you're using `2.4.x` :
1. Remove `nuxt-ts` dependency
2. Install `nuxt` and `@nuxt/typescript` dependencies
3. Remove your `tsconfig.json`, let Nuxt generate it with defaults at first run, then you can edit it to fit your needs.

If you're using `2.5.x` and have a working project, no changes are needed.

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly. (PR: https://github.com/nuxt/docs/pull/1269)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/cli/src/utils/typescript.js [hunk 1]
```diff
@@ -1,34 +1,16 @@
...
-    return false
-  }
-
-  consola.info(`${chalk.bold.blue('tsconfig.json')} found, enabling TypeScript runtime support`)
-
   try {
     const { setup } = require('@nuxt/typescript')
-    await setup(tsConfigPath, options)
+
+    consola.info('Initializing TypeScript Support')
```
Comment (pi0) [inline packages/cli/src/utils/typescript.js]: Can we show a message like "Creating /full/path/to/tsconfig.json" ?

```js
const tsConfigPath = path.resolve()
consola.info
await setup
```

**Comment by pi0** [comment]
@kevinmarrec I think we would also need a proper migration guide alongside with this PR changes

File: packages/cli/src/utils/typescript.js (hunk 1 above)
Comment (kevinmarrec) [reply packages/cli/src/utils/typescript.js]: What about just replace `tsconfig.json` by full path here ? : https://github.com/nuxt/nuxt.js/pull/5367/commits/fe3ecd63588390fdea30ba231c03a8a3eed4f78a#diff-8404199e6ca4961076ae35d7afeb980aR48

**Comment by kevinmarrec** [comment]
@pi0 yes sure, will only make the PR out of DRAFT when everything will be good

File: packages/cli/src/utils/typescript.js (hunk 1 above)
Comment (pi0) [reply packages/cli/src/utils/typescript.js]: Yeah. Cool.

**Comment by kevinmarrec** [comment]
Btw, I don't know what's that test unit error output, looks weird, gonna try to figure out why it's happening and why it's failing

**Comment by kevinmarrec** [comment]
Well I tried to find what's wrong with tests, but as my changes are not relevant to the error and that it works on `dev` branch, I'm really confused...

**Comment by kevinmarrec** [comment]
@pi0 **IMPORTANT** : I'd like to have a "Co-authored-by" #5356 owner in the merge commit message when it gets merged in `dev`

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
@kevinmarrec 9a3fc8a44e3cf3a8f551fa93ba60eb88a8c39663

**Comment by kevinmarrec** [comment]
Thanks @pi0

**Pull request merged**